docs/apis/batch/libs/table.md

```diff
@@ -67,6 +67,165 @@ The central concept of the Table API is a `Table` which represents a table with
 
 The following sections show by example how to use the Table API embedded in the Scala and Java DataSet APIs.
 
+### Registering Tables to and Accessing Tables from TableEnvironments
+
+`TableEnvironment`s have an internal table catalog to which tables can be registered with a unique name. After registration, a table can be accessed from the `TableEnvironment` by its name. Tables can be registered in different ways.
```

Comment:
Shall we add a comment that registering a table is only required for SQL queries and not for "pure" Table API programs?